python/pyspark/sql/readwriter.py

```diff
@@ -922,6 +925,8 @@ def csv(self, path, mode=None, compression=None, sep=None, quote=None, escape=No
                          the default UTF-8 charset will be used.
         :param emptyValue: sets the string representation of an empty value. If None is set, it uses
                            the default value, ``""``.
+        :param lineSep: defines the line separator that should be used for writing. If None is
+                        set, it uses the default value, ``\\n``. Maximum length is 1 character.
```

Comment:
Not sure if I'm missing something, but has this removed the ability use `\r\n`?

Reply:
Spark never supported `\r\n` in writing path.

Reply:
Revisiting this since I'd like to get rid of a local patch.

Why do you say it doesn't support this?

Reverting to the 2 character restriction works in my testing, on both the read and write paths and using arbitrary two character delimiters.

Reply:
Sorry for the extra comments: hadn't read deeply enough.

So the problem is Univocity's `normalizedNewLine` stuff? It fails in multiline cases? That's what I'm seeing in the tests and would explain why I don't see it in my use cases.

If that's the case, wondering if it's okay to allow two characters for the non-multiline cases?

Reply:
Please file a JIRA and go ahead if you can.

Reply:
ack